Sources/SmithyTestUtil/CBORComparator.swift

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+import AwsCommonRuntimeKit
+import Foundation
+@_spi(SmithyReadWrite) import SmithyCBOR
+
+public struct CBORComparator {
```

Comment:
this is solely used for testing. we need to compare the semantic values rather than the exact binary values. CRT should have sufficiently tested their de-serializer against SEP prescribed tests to ensure that using the Decoder in protocol test comparison is ok

Reply:
Is the testing in CRT side done already? If not, what's the known timeline?